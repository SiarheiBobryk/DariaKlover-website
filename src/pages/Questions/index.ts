export { default } from './Questions';
export { default as questionsMetaData } from './questionsMetaData';
