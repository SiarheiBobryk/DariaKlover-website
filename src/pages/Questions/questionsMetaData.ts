export interface QuestionMetaData {
  title: string;
  heading: string;
  path: string;
}

const questionsMetaData: QuestionMetaData = {
  title: 'Задать вопрос',
  heading: 'Вопросы к Дарье',
  path: '/question',
};

export default questionsMetaData;
