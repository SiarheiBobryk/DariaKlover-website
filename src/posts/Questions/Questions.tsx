import * as React from 'react';

import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

import Paragraph from '../../components/Paragraph';
import Heading from '../../components/Heading';

export interface QuestionsProps {
  heading?: string;
}

function Questions(props: QuestionsProps) {
  const { heading } = props;

  return (
    <>
      <Heading>{heading}</Heading>
      <Paragraph>–ó–¥–µ—Å—å –º–æ–∂–Ω–æ –±—É–¥–µ—Ç –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å –î–∞—Ä—å–µ</Paragraph>
      <Alert sx={{ my: 1 }} severity="info">
        <AlertTitle>–ö–∞–∫ –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å</AlertTitle>
        –§–æ—Ä–º–∞, –≥–¥–µ –º–æ–∂–Ω–æ –±—É–¥–µ—Ç –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å, –∞ —Ç–∞–∫–∂–µ –ø–æ—Å–º–æ—Ç—Ä–µ—Ç—å —á–∞—Å—Ç–æ –∑–∞–¥–∞–≤–∞–µ–º—ã–µ –≤–æ–ø—Ä–æ—Å—ã, —Å–∫–æ—Ä–æ –±—É–¥–µ—Ç –æ–ø—É–±–ª–∏–∫–æ–≤–∞–Ω–∞ –Ω–∞
        —ç—Ç–æ–π —Å—Ç—Ä–∞–Ω–∏—Ü–µ. –°–ø–∞—Å–∏–±–æ –∑–∞ –≤–∞—à–µ —Ç–µ—Ä–ø–µ–Ω–∏–µ üòá
      </Alert>
    </>
  );
}

export default Questions;
