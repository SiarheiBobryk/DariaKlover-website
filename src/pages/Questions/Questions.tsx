import * as React from 'react';
import { Helmet } from 'react-helmet';

import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

import Heading from '../../components/Heading';
import LinkBlank from '../../components/LinkBlank';
import Paragraph from '../../components/Paragraph';
import questionsMetaData from './questionsMetaData';

function Questions() {
  return (
    <>
      <Helmet>
        <title>{questionsMetaData.title}</title>
      </Helmet>
      <Heading>{questionsMetaData.heading}</Heading>
      <Paragraph>Пожалуйста, перейдите по ссылке ниже. Там будет форма, чтобы задать вопрос Дарье.</Paragraph>
      <Paragraph>
        {/* eslint-disable-next-line max-len */}
        <LinkBlank href="https://docs.google.com/forms/d/e/1FAIpQLSedaIyUajfTsT7ymkNJd8umXnUV-Tc2q6OTmw3Xk8NGB9Hs0A/viewform?usp=sf_link">
          Вопрос Дарье Кловер
        </LinkBlank>
      </Paragraph>
      <Alert severity="warning">
        <AlertTitle>Система тестируется</AlertTitle>
        Обращаем ваше внимание, что страничка, где можно задать вопрос, находится в режиме тестирования и может быть
        нестабильна.
      </Alert>
    </>
  );
}

export default Questions;
