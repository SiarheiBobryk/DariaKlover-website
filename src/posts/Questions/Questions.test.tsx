import * as React from 'react';

import { render, RenderResult } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { axe, toHaveNoViolations } from 'jest-axe';

import Questions from './Questions';

expect.extend(toHaveNoViolations);

describe('<Questions />', () => {
  describe('conformance', () => {
    it('should match a11y standards', async () => {
      const { container }: RenderResult = render(<Questions heading="Lorem Ipsulum" />);

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  describe('snapshotting', () => {
    it('should render correctly', () => {
      const questions: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<Questions />).toJSON();
      expect(questions).toMatchSnapshot();
    });
  });
});
