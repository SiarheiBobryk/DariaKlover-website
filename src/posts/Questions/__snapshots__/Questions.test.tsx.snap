// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Questions /> snapshotting should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  />,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ó–¥–µ—Å—å –º–æ–∂–Ω–æ –±—É–¥–µ—Ç –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å –î–∞—Ä—å–µ
  </p>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-efxmhs-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
        />
      </svg>
    </div>
    <div
      className="MuiAlert-message css-acap47-MuiAlert-message"
    >
      <div
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        –ö–∞–∫ –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å
      </div>
      –§–æ—Ä–º–∞, –≥–¥–µ –º–æ–∂–Ω–æ –±—É–¥–µ—Ç –∑–∞–¥–∞—Ç—å –≤–æ–ø—Ä–æ—Å, –∞ —Ç–∞–∫–∂–µ –ø–æ—Å–º–æ—Ç—Ä–µ—Ç—å —á–∞—Å—Ç–æ –∑–∞–¥–∞–≤–∞–µ–º—ã–µ –≤–æ–ø—Ä–æ—Å—ã, —Å–∫–æ—Ä–æ –±—É–¥–µ—Ç –æ–ø—É–±–ª–∏–∫–æ–≤–∞–Ω–∞ –Ω–∞ —ç—Ç–æ–π —Å—Ç—Ä–∞–Ω–∏—Ü–µ. –°–ø–∞—Å–∏–±–æ –∑–∞ –≤–∞—à–µ —Ç–µ—Ä–ø–µ–Ω–∏–µ üòá
    </div>
  </div>,
]
`;
